src/main/java/org/ladocuploader/app/preparers/MedicalExpensesPreparer.java

```diff
@@ -0,0 +1,54 @@
+package org.ladocuploader.app.preparers;
+
+import formflow.library.data.Submission;
+import formflow.library.pdf.PdfMap;
+import formflow.library.pdf.SingleField;
+import formflow.library.pdf.SubmissionField;
+import formflow.library.pdf.SubmissionFieldPreparer;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+
+@Component
+@Slf4j
+public class MedicalExpensesPreparer implements SubmissionFieldPreparer {
+  private static final Map<String, String> EXPENSES = new HashMap<>();
+
+  static {
+    EXPENSES.put("householdMedicalExpenseAmount_wildcard_Dental bills", "Dental bills");
+    EXPENSES.put("householdMedicalExpenseAmount_wildcard_Hospital bills", "Hospital bills");
+    EXPENSES.put("householdMedicalExpenseAmount_wildcard_Prescribed medicine", "Prescribed medicine");
+    EXPENSES.put("householdMedicalExpenseAmount_wildcard_Prescription drug plan premium", "Prescription drug plan premium");
+    EXPENSES.put("householdMedicalExpenseAmount_wildcard_Medical appliances", "Medical appliances");
+    EXPENSES.put("householdMedicalExpenseAmount_wildcard_Health insurance or Medicare premiums", "Health insurance or Medicare premiums");
+    EXPENSES.put("householdMedicalExpenseAmount_wildcard_Nursing home", "Nursing home");
+    EXPENSES.put("householdMedicalExpenseAmount_wildcard_Other medical expenses", "Other medical expenses");
+  }
+
+  @Override
+  public Map<String, SubmissionField> prepareSubmissionFields(Submission submission, PdfMap pdfMap) {
+    Map<String, SubmissionField> results = new HashMap<>();
+
+    var expenses = (List) submission.getInputData().getOrDefault("householdMedicalExpenses[]", emptyList());
+    if (!expenses.isEmpty()) {
+      List<String> sortedExpenses = EXPENSES.keySet().stream().sorted().toList();
```

Comment:
Might be worth keeping another static class variable that contains this value - so you can just do it once overall and not on every call to this.

Reply:
For the sorting part - I can see why you need a list of keys here, but sort still may not be necessary.

Reply:
This raises an interesting question for me -- what if you wanted to sort a preparer list - like the medical one. Are you finding that if you sort and then add fields in that order they show up in that order?